elastic/kibana
# [Infra UI] Disable Osquery in serverless

closes https://github.com/elastic/kibana/issues/163280

## Summary

Disables OSquery in serverless. 


<img width="1670" alt="image" src="https://github.com/elastic/kibana/assets/2767137/0ece0503-a5f3-46a1-a99b-29285088526c">

<img width="1670" alt="image" src="https://github.com/elastic/kibana/assets/2767137/5d20aaa4-7bdc-47e5-b9c2-469ecf62ab1f">


<img width="1670" alt="image" src="https://github.com/elastic/kibana/assets/2767137/97874aee-be98-4f0a-984a-562abbd4187d">

Osquery is now only used in the Asset Details flyout/full-page view. The header component in the Asset Details now checks the feature flag object to show/hide the OSquery tab.

### How to test

- Start a local es instance: `yarn es serverless --kill --clean --license trial --ssl`
- Enable `infra` in the `serverless.oblt.dev.yml` file:
  - `xpack.infra.enabled: true`
- Start a local kibana instance: `yarn serverless-oblt --ssl` and see if the side nav contains the Infrastructure item
- Navigate to `https://0.0.0.0:5601/ftw/app/metrics` and open the Asset Details flyout from Inventory and Hosts View. Navigate to the details view and check that Osquery tab isn't present.

- Confirm if OSquery tab is present in stateful.
